### PR TITLE
CI: Switch to Github Actions

### PR DESCRIPTION
**Pull request by sgallagher**
Drop all integration with Travis CI and replace it with Github
Actions. Among other things, the Github Actions will run more
tests in parallel, so there should be a much shorter turnaround on
CI test results.

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

### CODE REVIEW

File: .ci/coverity/ci-tasks.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ meson --buildtype=debug \
 pushd coverity
 
 # The coverity scan script returns an error despite succeeding...
- TRAVIS_BRANCH="${TRAVIS_BRANCH:-main}" \
+ TRAVIS_BRANCH="main" \
```
Comment (mmathesius) [inline .ci/coverity/ci-tasks.sh:28]: Does coverity still require references to travis? Even the script below is named `**travis**ci_build_coverity_scan.sh`.

**Review by mmathesius — CHANGES_REQUESTED** [review]
Overall, this looks good! Thanks for doing this. I'll have to learn more about Github Actions one of these days.

`.gitignore` should have lines added for `.ci/.home*` and
`.ci/*/Dockerfile.deps.*`

Additional notes: the renamed `.ci/centos/ci-tasks.sh` and `.ci/coverity_prep.sh` files still have references to travis.

File: fedora/libmodulemd.spec [hunk 2]
```diff
@@ -16,13 +16,13 @@
 %define _lto_cflags %{nil}
 
 Name:           %{upstream_name}%{?v2_suffix}
-Version:        0.0.0
-Release:        0
+Version:        2.10.1
+Release:        0.20201209202157820929.github_actions.10.g775d652%{?dist}
```
[the remark this answers is not in the document]
Comment (sgallagher) [reply fedora/libmodulemd.spec:20]: No

**Comment by sgallagher** [comment]
> Overall, this looks good! Thanks for doing this. I'll have to learn more about Github Actions one of these days.
> 
> `.gitignore` should have lines added for `.ci/.home*` and
> `.ci/*/Dockerfile.deps.*`
> 
> Additional notes: the renamed `.ci/centos/ci-tasks.sh` and `.ci/coverity_prep.sh` files still have references to travis.

The `coverity_prep.sh` needs to, because I'm still using their script that's designed for (and named after) travis.

File: .ci/coverity/ci-tasks.sh (hunk 1 above)
Comment (sgallagher) [reply .ci/coverity/ci-tasks.sh:28]: Yes, the script we pull from scan.coverity.com requires TRAVIS_BRANCH.

**Review by mmathesius — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**